target/linux/ramips/image/mt7621.mk

```diff
@@ -1243,6 +1243,22 @@ define Device/xiaomi_mi-router-3-pro
 endef
 TARGET_DEVICES += xiaomi_mi-router-3-pro
 
+define Device/xiaomi_mi-router-4
```

Comment:
And you are perfectly sure that you don't need uimage-lzma-loader include here?

dsa-migration is missing as well, which tells me you didn't test sysupgrade ...